applications/tari_console_wallet/src/ui/state/app_state.rs

```diff
@@ -441,6 +449,9 @@ pub struct AppStateInner {
     updated: bool,
     data: AppStateData,
     wallet: WalletSqlite,
+    balance_enquiry_cooldown_start: Instant,
+    balance_enquiry_cooldown_infringement: bool,
```

Comment:
This is a real over complication of a simple feature. Adding a lot of code to just produce a trace log of when this occurs is really unnecessary.

Reply:
Is this in place to combat the edge case I mentioned in the review?

Reply:
The purpose here is not a trace log. As you have observed, all requests in the cooldown period after the last accepted request will be ignored. To ensure that the last request in a cooldown period is processed if no more requests are made after that, we need a separate event to trigger such a check, such that the last ignored request in a cooldown period will be handled.

To summarize:
- regular calls to calculate the balance ends up with `refresh_balance()`;
- `refresh_balance()` will only perform the balance calculation if there is no cooldown period infringement and clear the `balance_enquiry_cooldown_infringement` flag, otherwise, it will set the `balance_enquiry_cooldown_infringement` flag;
- `refresh_balance_check()`, which runs in a separate thread with the regular `BaseNodeEvent::BaseNodeStateChanged`, will call upon `refresh_balance()` if the `balance_enquiry_cooldown_infringement` flag has been set.